File: shared/schemas/case.py
from typing import Any
from pydantic import BaseModel


class CaseResponse(BaseModel):
    case_id: str
    customer_id: str | None = None
    kyc_risk: float | None = None
    aml_risk: float | None = None
    fused_risk: float | None = None
    uncertainty: float | None = None
    priority: str | None = None
    human_review_required: bool = False
    suspected_typology: str | None = None
    shap_features: list[dict[str, Any]] | None = None
    graph_explanation: dict[str, Any] | None = None
    status: str

    class Config:
        from_attributes = True


class CaseStatusUpdate(BaseModel):
    status: str

class CaseFeedback(BaseModel):
    analyst_decision: str
    comment: str | None = None

File: apps/frontend/src/types/case.ts
export interface ShapFeature {
  feature: string;
  value: number;
  importance: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  transaction_id: string;
  importance: number;
}

export interface GraphExplanation {
  nodes: string[];
  edges: GraphEdge[];
  important_features: Array<{ feature: string; importance: number }>;
}

export interface CaseRecord {
  case_id: string;
  customer_id: string | null;
  kyc_risk: number | null;
  aml_risk: number | null;
  fused_risk: number | null;
  uncertainty: number | null;
  priority: string | null;
  human_review_required: boolean;
  suspected_typology: string | null;
  shap_features: ShapFeature[] | null;
  graph_explanation: GraphExplanation | null;
  status: string;
}

export type CaseStatus = "OPEN" | "UNDER_REVIEW" | "CLOSED";
export type AnalystDecision =
  | "CONFIRMED"
  | "FALSE_POSITIVE"
  | "NEEDS_MORE_INVESTIGATION";

export interface FeedbackPayload {
  analyst_decision: AnalystDecision;
  comment: string | null;
}
